### ISSUE TITLE
Re-create sub tasks for repeating tasks

### ISSUE THREAD

**Issue by level6sam** [johannesjo/super-productivity#427]
### Your Environment

* Version used: snap
* Operating System and version: Ubuntu 18.04
* Desktop Environment:  Elementary OS 5.1.4
* Browser Name and version: 

### Expected Behavior

Create a main task for items that need to be carried out
Make the main task repeatable so that it appears the next day
Create subtasks under the main task
Schedule the tasks for the next day

### Current Behavior

Open up the app the day after scheduling a sub-task from a repeatable main task
Main task appears but there isn't a sub task anymore making the main task useless

### Steps to Reproduce (for bugs)

1. Create a main task
2. Add sub tasks
3. Make the main task repeatable and save
4. Wait the next day to find the sub tasks no longer appear under the main task

**Comment by johannesjo** [comment]
Hey! Thanks for opening this up! This is actually a conscious decision, but I can see that this depends on the actual use case. I leave this open for discussion.

**Comment by level6sam** [comment]
So my argument (for discussion) is that sub tasks should be a repeatable option because much of the work we carry out is mainly through schemas. With the ability to monitor the amount of time spent compared to how much time should be spent. It helps motivate a person who knows that these repeatable tasks are uninspiring and are more or less chores. Eventually the idea would be to have done it enough times for it to become a (healthy) habit it itself. Also, it is a pain to constantly re-order tasks which get mixed up in other one off tasks as its repeatable tasks usually follow a specific order in which they are completed.

Thanks

**Comment by nicknovitski** [comment]
> Hey! Thanks for opening this up! This is actually a conscious decision, but I can see that this depends on the actual use case.

I can understand that.  What I imagine is, as part of the Edit Repeat Task Config window, you can optionally specify default sub-tasks which will be created when the task is automatically created, and completed when the task is automatically marked completed.  So that people can mix default sub-tasks with manually added ones, which can retain the previous behavior.

My use case is: I want a daily reminder to empty my inboxes, and I want to work on each different inbox (for example email, slack, github) one at a time, usually in the same order.   Essentially, I'd like to be able to create repeating tasks that function as checklists.

**Comment by johannesjo** [comment]
I am open for experimenting with this, if someone is willing to put work into it :)

**Comment by aninternetuserx** [comment]
In case it's useful for someone, since this feature is not available right now, the workaround I found is to use the "comment" section of the repeating task to add the little steps of the task and they repeat with it :)

**Comment by podivilov** [comment]
Hey @johannesjo, will this feature be added in future release?

**Comment by aninternetuserx** [comment]
I think he already answered that @podivilov  
> I am open for experimenting with this, if someone is willing to put work into it :)

**Comment by podivilov** [comment]
> I think he already answered that @podivilov
> 
> > I am open for experimenting with this, if someone is willing to put work into it :)

Oops!

**Comment by Sundjata** [comment]
I've started using the app more consistently and this is a killer issue for me. The repeating tasks really aren't useful to me without the subtasks attached and the ability to track time during each sub task. I want to know which step of the process takes the most time and I want to track that over an extended period of time. The alternative would be to only use parent tasks and to create separate projects to organize each set of tasks into. I'm trying to avoid this as it would make evaluating projects a lot more daunting. I don't know much about coding but I love the app and would be willing to learn if it'll bring us closer to having this feature. How difficult would this be to implement?

**Comment by Cyber-Syntax** [comment]
I think this feature should be in the app because it's a really good and efficient way to deal with big tasks. For example, I'm trying to learn python and it's a big job. I need to add some sub tasks each week to "learn python" like functions, loops, exceptions and that will give me how much time I can spend with each section. Also, it's gonna save me to struggling with each tasks for function, loops etc.

**Comment by frenchthorn** [comment]
Same, aaylnx! I wish I could make this happen, but my coding skill is approximately zero.

**Comment by aaylnx** [comment]
Yes. As great as SP is, this would be a tremendous improvement. Anyone with the skills able to assist @johannesjo with this?

**Comment by frainz-de** [comment]
Yes, the issue is still relevant.

**Comment by mwaterbu** [comment]
As a new user, this feature would make this app nearly perfect for me. It would be very helpful for tracking things like daily routines and chore lists, which are much better under a single task, rather than cluttering the list with 10-20 tasks.

**Comment by mwaterbu** [comment]
Still relevant

**Comment by alooful** [comment]
Still relevant. You can't repeat sub-tasks? What is the alternative? Having your routines de-organized? I guess I can find a way to work around this.

**Comment by CodeObserver21** [comment]
I ran into the same issue. I read some of the tutorials and guides for Super Productivity. While repeating tasks can't have subtasks, they can still have notes. So for the meantime, you can make a repeating task and write the subtasks into the notes section using markdown:

\- [X] Check emails
\- [ ] Make some coffee
\- [ ] Go for a walk

And if you have the setting to "Render notes into markdown" turned on, it will look like this in the notes section:

- [X] Check emails
- [ ] Make some coffee
- [ ] Go for a walk

I've confirmed that the notes of a repeated task instance are independent of the source and won't change the template you made. Only small caveat is that you can't click on the box to check-mark it. You'll have to manually edit the note and put an "X". Not too bad of a work around. For my use cases, this has worked out just fine for the moment.

**Comment by aeshna-cyanea** [comment]
still relevant!

**Comment by inattentive-nerd** [comment]
It would be nice having an option for subtasks to be repeated with the parent task or not.

**Comment by mattastic485** [comment]
Would love to have this feature as well.

**Comment by KyanoAng3l0** [comment]
I hope someone implements this. I'm recently diagnosed with ADHD (in my 30s 😐) and have turned to apps such as this as aids since I'm on the computer most of the time. I'm told to break down tasks, and it does help a lot.

**Comment by Pinkolik** [comment]
Hello. I think this would be a nice feature. I think subtasks should have an option to be repeatable independently of their parent. And also subtasks are very nice for more precise time tracking

**Comment by Baije7Sensai** [comment]
> I hope someone implements this. I'm recently diagnosed with ADHD (in my 30s 😐) and have turned to apps such as this as aids since I'm on the computer most of the time. I'm told to break down tasks, and it does help a lot.

Same!

> Hello. I think this would be a nice feature. I think subtasks should have an option to be repeatable independently of their parent. And also subtasks are very nice for more precise time tracking

Yes, please!

**Comment by rjfnyc** [comment]
I would like the feature to have sub-tasks of repeating tasks to be repeated along with parent task.

**Comment by jalbersdorfer** [comment]
Yes, I also would **very appreciate to have the Subtasks of repeating Tasks would re-create** with the Main-Task every Schedule.
I will give the **Checklist**-Feature a try, but it is by far not as comfortable as repeating Subtasks would be.
Thank you in advance. Otherwise a very handy Productivity Tool 👍 which I really love 😃 to use every day.

**Comment by fabiante** [comment]
I'd also like to have subtasks be recreated. The checkbox in the markdown desc. does not allow time tracking which Is what I need for the subtasks.

A workaround would be to not use subtasks and instead have N main tasks be repeated. I personally don't want to do that because I then lack the parent-child-aggregation in the daily summary.

I suggest that this could be an option when setting up the repetition? A checkbox "Include subtasks"?

**Comment by Baije7Sensai** [comment]
Don't bother, people have been asking for this feature for couples years now, but the developers just keep ignoring that...

Honestly, it's a deal breaker for me after seeing how long people have been asking for this feature, but the developer being adamant that's it better not to include it for the users who wants it...
I just deleted the tool and looked elsewhere

**Comment by johannesjo** [comment]
@Baije7Sensai it is not true that I am ignoring this request, there is just a lot of stuff to do and since this is open source everyone can contribute. It even might be a good thing if this was implemented by someone who really wants this feature personally rather than by me.

**Comment by suraj-mandal** [comment]
Has this been implemented?

**Comment by Universal-Igloo** [comment]
@johannesjo Please do implement this if you are able to 🙏🏽 You can make it a toggle-able option in settings if you so desire, clearly there are quite a few people who expect repeated tasks to function this way. I second the experiences of other commenters with ADHD, it makes a huge difference for neurodiverse people. Thank you for the great app! 🥰

**Comment by suraj-mandal** [comment]
@johannesjo , I would like to have a look into this issue.. if no one is actively working on it yet!

**Comment by johannesjo** [comment]
That would be great @suraj-mandal ! 

There is an open but unfished PR for this btw.: https://github.com/johannesjo/super-productivity/pull/2659

Maybe some of it is helpful?

**Comment by suraj-mandal** [comment]
Sure, I am looking forward to this

**Comment by suraj-mandal** [comment]
I will be taking around 3 days of time so that I go through the entire codebase and understand what it is doing, so that I can work on this accordingly. If someone can work on this meanwhile, they can, otherwise, once I am sure which is 3 days, I will start picking this up. Thanks!

**Comment by johannesjo** [comment]
Take all the time you need @suraj-mandal ! There is no rush!

**Comment by ReduxST** [comment]
This is a good enhancement suggestion and I found it missing almost immediately when I started using this a few days ago. 
I didn't want to wait for Suraj to do it as its been a while and they might have forgotten.

I've tried implementing it in the same way the parent repeating task is handled. My local implementation is working as expected from the testing I've done. Because not everyone will prefer the sub-tasks repeating, I've made it toggle-able, default off.
I'll properly fork the repo, commit my local changes, and submit a pull request with the changes soon, likely within the next day or so.

**Comment by suraj-mandal** [comment]
Hi @ReduxST , sure.. I was bit busy this time and could not find much time.. you can carry on with this..

**Comment by pclahoud** [comment]
Happy to see that this is being worked on! Any idea on when this will be added to main?

**Comment by johannesjo** [comment]
This should be already released. I think we should further improve the UI, but the basic feature is there.

**Comment by mwaterbu** [comment]
> This should be already released. I think we should further improve the UI, but the basic feature is there.

I wasn't aware that this had been completed. While I agree that the current UI is basic, it now does exactly what I needed it to do! Thanks to everyone who helped with this!

**Comment by BerkhanB** [comment]
Is it also possible to have the same feature for sub-tasks as well?
